BedrockCommand.cpp

```diff
@@ -127,6 +130,83 @@ bool BedrockCommand::areHttpsRequestsComplete() const {
     return true;
 }
 
+void BedrockCommand::_waitForHTTPSRequests() {
+    uint64_t startTime = 0;
+    while (!areHttpsRequestsComplete()) {
+        // Wait until the command's timeout, or break early if the command has timed out.
+        uint64_t maxWaitUs = 0;
+        uint64_t now = STimeNow();
+        if (!startTime) {
+            startTime = now;
+        }
+        if (now < timeout()) {
+            maxWaitUs = timeout() - now;
+        } else {
+            // This uses the same starting point as in areHttpsRequestsComplete, for efficiency.
+            // We won't iterate over large numbers of known completed requests, instead starting
+            // from the the point of the list of known completed request.
+            auto requestIt = (_lastContiguousCompletedTransaction == httpsRequests.end()) ? httpsRequests.begin() : _lastContiguousCompletedTransaction;
+            while (requestIt != httpsRequests.end()) {
+                if (!(*requestIt)->response) {
+                    (*requestIt)->response = 500;
```

Comment:
nab, it would be nice to make this a non-standard code so we know its an internal bedrock error response. we have had a lot of confusion in the past where people think external services are responding `500` but really bedrock is. making this something like `1050` or some other non-standard http code would improve log readability.

Reply:
I thought of making it return 504 (which is still a standard error code) but decided to use 500 because it's what we already use [here](https://github.com/Expensify/Bedrock/blob/4278b1f0e55df46be9441ba2ef0f43a99e1bc9ba/libstuff/SHTTPSManager.cpp#L138) and I didn't want to break anything that expects that.